### PR TITLE
VIRTS-1208

### PR DESCRIPTION
**Pull request by garunagiri**
All objects have a 'hidden' field since it's now in base_object. Only operations and adversaries can have this set to true though for now.

*Extended to abilities as well

### CODE REVIEW

File: app/service/rest_svc.py [hunk 1]
```diff
@@ -234,8 +234,7 @@ async def _build_operation_object(self, access, data):
         name = data.pop('name')
         group = data.pop('group', '')
         planner = await self.get_service('data_svc').locate('planners',
-                                                            match=dict(name=data.pop('planner') if not
-                                                                       data.get('planner') == '' else 'sequential'))
+                                                            match=dict(name=data.pop('planner', 'sequential')))
```
Comment (khyberspache) [inline app/service/rest_svc.py:237]: This won't work.  The data.pop will check if it's NULL.  UI returns an empty string, not None.

Comment (garunagiri) [reply app/service/rest_svc.py:237]: Ah okay, my bad. Thanks! I'll revert that

File: app/service/data_svc.py [hunk 2]
```diff
@@ -135,6 +135,7 @@ async def _load_abilities(self, plugin):
                     privilege = ab.pop('privilege', None)
                     repeatable = ab.pop('repeatable', False)
                     requirements = ab.pop('requirements', [])
+                    hidden = ab.pop('hidden', False)
```
Comment (scottctaylor12) [inline app/service/data_svc.py:138]: Move the your _load_abilities changes to the new load_ability_file function found on the master branch. That will resolve the merge conflict.

File: app/service/data_svc.py [hunk 3]
```diff
@@ -158,7 +159,8 @@ async def _load_abilities(self, plugin):
                                                                    requirements=requirements, privilege=privilege,
                                                                    buckets=await self._classify(ab, tactic),
                                                                    access=plugin.access, repeatable=repeatable,
-                                                                   variations=info.get('variations', []), **ab)
+                                                                   variations=info.get('variations', []), hidden=hidden,
```
Comment (scottctaylor12) [inline app/service/data_svc.py:162]: Same as above, move to load_ability_file function

File: app/objects/c_ability.py [hunk 4]
```diff
@@ -64,7 +65,7 @@ def unique(self):
     def __init__(self, ability_id, tactic=None, technique_id=None, technique=None, name=None, test=None,
                  description=None, cleanup=None, executor=None, platform=None, payloads=None, parsers=None,
                  requirements=None, privilege=None, timeout=60, repeatable=False, buckets=None, access=None,
-                 variations=None, language=None, code=None, build_target=None, **kwargs):
+                 variations=None, language=None, code=None, build_target=None, hidden=False, **kwargs):
```
Comment (scottctaylor12) [inline app/objects/c_ability.py:68]: Check out the get_variations function in the master branch to fix merge conflicts

**Comment by scottctaylor12** [comment]
@privateducky can you give a final review for this PR

**Comment by garunagiri** [comment]
Hidden right now is only used for the automatic red operations started by blue training certs. The training plugin will have abilities and adversaries that (I think) should be hidden from being viewed in the front end. The resulting operation will also be kept from being visible. This helps keep answers from being given away really easily. Though of course, if you look in the back they'll be there...

**Comment by scottctaylor12** [comment]
@privateducky can you do a final review of Govardhen's hidden adversary PR? This will be needed for the blue training certificate to seamlessly execute adversaries without user interaction. This is a requirement for a different PR in the training plugin: https://github.com/mitre/training/pull/75

**Comment by garunagiri** [comment]
Closing this, the issue is now being addressed by https://github.com/mitre/caldera/pull/1734

**Pull request closed without merging**